rockcraft/errors.py

```diff
@@ -0,0 +1,39 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2022 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Rockcraft error definitions."""
+
+from craft_cli import CraftError
+
+
+class RockcraftInitError(CraftError):
+    """Error while initializing rockcraft project."""
+
+
+class PartsLifecycleError(CraftError):
+    """Error during parts processing."""
+
+
+class ProviderError(CraftError):
+    """Error in provider operation."""
+
+
+class ProjectLoadError(CraftError):
+    """Error loading rockcraft.yaml."""
+
+
+class ProjectValidationError(CraftError):
+    """Error validatiing rockcraft.yaml."""
```

Comment:
```suggestion
    """Error validating rockcraft.yaml."""
```

Reply:
uff :stuck_out_tongue: I still need to find a proper VSCode extension for grammar...one that doesn't complain about all the programming variables, but notices proper English mistakes

thanks